kizzanaome/nomaCakes
# Create a client account

### What does this PR do?
 This PR is to signup a new user to the nomaCakes project.

### Description of Task to be completed?
 When a user or client wants to access some of the app's features like purchasing or adding to the cart, they have to create the account and this PR adds that to the app.
#### Signup user
`/api/authentication/signup`

### How should this be manually tested?
- Fetch this branch by running the command `git fetch origin ft-signup` in the terminal
- Check into this branch by running `git checkout ft-signup` in the terminal
- Activate your virtual environment by running `source nomaenv/bin/activate` in the terminal
- Create a database by typing `CREATE DATABASE <database name>;` preferably in a `psql` terminal.
- Create a .env file and use the sample env file to fill in the details required to connect to your database
- Apply the new changes to the models by running `python manage.py migrate` in the terminal
- Start the application by running `python manage.py runserver`

- load up postman and access the url `http://127.0.0.1:8000/api/authentication/signup`
- Add this payload and send the request:
```
{
	"username": "john doe",
	"contact": "0755574507",
	"email": "john@nomacakes.com",
	"password": "johndoe",
	"password2": "johndoe"
}
```

### Relevant screenshots
<img width="788" alt="Screen Shot 2020-01-27 at 9 11 24 PM" src="https://user-images.githubusercontent.com/34310682/73201264-b434d200-4149-11ea-8d65-3378d69c196f.png">
